### PR TITLE
visp: init at 3.6.0

### PR DESCRIPTION
**Pull request by nim65s**
## Description of changes

Visp is an open source visual servoing library: https://visp.inria.fr/

Nix support & CI was added upstream in https://github.com/lagadic/visp/pull/1416.
This PR is based on that work, plus darwin fixes.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by nim65s** [comment]
Result of `nixpkgs-review pr 333473` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>visp</li>
  </ul>
</details>

**Comment by nim65s** [comment]
Result of `nixpkgs-review pr 333473` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>visp</li>
  </ul>
</details>

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

File: pkgs/by-name/vi/visp/package.nix [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    # this was merged upstream, and can be removed on next release
+    (fetchpatch {
+      url = "https://github.com/lagadic/visp/pull/1416/commits/fdda5620389badee998fe1926ddd3b46f7a6bcd8.patch";
+      hash = "sha256-W3vvBdQdp59WAMjuucNoWI0eCyPHjWerl7VCNcPVvzI=";
+    })
+
+    # fix for opencv new Universal Intrinsic API
+    # this was merged upstream, and can be removed on next release
+    (fetchpatch {
+      url = "https://github.com/lagadic/visp/pull/1310/commits/9ed0300507e13dddd83fd62a799f5039025ea44e.patch";
```
[the remark this answers is not in the document]
Comment (nim65s) [reply pkgs/by-name/vi/visp/package.nix:55]: Why would we want that ? I think having the pull request reference is a good thing.

File: pkgs/by-name/vi/visp/package.nix [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "visp";
+  version = "3.6.0";
+
+  src = fetchFromGitHub {
+    owner = "lagadic";
+    repo = "visp";
+    rev = "v${finalAttrs.version}";
```
[the remark this answers is not in the document]
Comment (nim65s) [reply pkgs/by-name/vi/visp/package.nix:40]: Why ?

File: pkgs/by-name/vi/visp/package.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (FliegendeWurst) [reply pkgs/by-name/vi/visp/package.nix:55]: You can always include the PR in the comment above. I suppose it doesn't matter whether the url contains pull/xxx..

File: pkgs/by-name/vi/visp/package.nix (hunk 2 above)
[the remark this answers is not in the document]
Comment (FliegendeWurst) [reply pkgs/by-name/vi/visp/package.nix:40]: If there is a branch with the same name, the fetch would fail.

**Review by FliegendeWurst — APPROVED** [review]

**Comment by nim65s** [comment]
@ofborg build visp

**Pull request merged**